lib/matplotlib/rcsetup.py

```diff
@@ -532,6 +536,50 @@ def validate_ps_distiller(s):
                          'ghostscript or xpdf')
 
 
+# A validator dedicated to the named line styles, based on the items in
+# ls_mapper, and a list of possible strings read from Line2D.set_linestyle
+_validate_named_linestyle = ValidateInStrings(
+    'linestyle',
+    [*ls_mapper.keys(), *ls_mapper.values(), 'None', 'none', ' ', ''],
+    ignorecase=True)
+
+
+def _validate_linestyle(ls):
+    """
+    A validator for all possible line styles, the named ones *and*
+    the on-off ink sequences.
+    """
+    if isinstance(ls, str):
+        try:  # Look first for a valid named line style, like '--' or 'solid'.
+            return _validate_named_linestyle(ls)
+        except ValueError:
+            pass
+        try:
+            ls = ast.literal_eval(ls)
```

Comment:
Is this only there to support reading rc_files? If so, I would mention this in a comment.

Reply:
That's literally the case for *all* functions in rcsetup.py...

Reply:
But most of them work implicitly because the don't parse nested structures. Then `str` -> other type is trivial.

Actually, I'd be in favor of making rcfiles YAML compliant. They mostly are already. We could delegate all the string parsing to the YAML parser. Then, the validators would only operate on the actual python types (not on their string representation). This would also be beneficial for using validators when setting rcParams directly in the code. But that's for another time.

Reply:
Added a comment.

I am strongly opposed to using yaml here; it's the wrong tool for the task: it's not expressive enough to describe things that are actually problematic in the current rc syntax (axes.prop_cycler, path.effects), while bringing in a lot of complexity (a typical one being the many, many different kind of strings that exist) that is unwarranted.

Reply:
`axes.prop_cycle` could well be described by a dict
~~~
axes.prop_cycle:
    color: ['1f77b4', 'ff7f0e', '2ca02c']
    linestyle: ['-', '--', ':']
~~~
But no need to go into that discussion here.